cli/src/lib.rs

```diff
@@ -50,25 +51,95 @@ pub struct Arguments {
     /// Set this flag on the peer that should submit genesis on the network initial start.
     pub submit_genesis: bool,
     /// Set custom genesis file path. `None` if `submit_genesis` set to `false`.
-    pub genesis_path: Option<PathBuf>,
+    pub genesis_path: Option<ConfigPath>,
     /// Set custom config file path.
-    pub config_path: PathBuf,
+    pub config_path: ConfigPath,
 }
 
-const CONFIGURATION_PATH: &str = "config.json";
-const GENESIS_PATH: &str = "genesis.json";
+const CONFIGURATION_PATH: &str = "config";
+const GENESIS_PATH: &str = "genesis";
+const PRECONFIGURED_CONFIG_EXTENSIONS: [&str; 2] = ["json", "json5"];
```

Comment:
A bit unclear why "preconfigured" instead of "supported" or "default". Not necessarily insisting on latter, just don't understand your what you mean by "preconfigured"

Reply:
Or maybe "allowed" would pass as well, though not insisting either.

Reply:
Okay, I like *default*

Reply:
Or *allowed* if we disallow other extensions